### PR TITLE
Update xgboost4j and include xgboost runtime in databricks runtime

### LINKED ISSUE #562: [xgboost][databricks][scala] key not found xgboost.regression
Hi, I was wondering if this is a known issue or if I'm doing something obviously wrong.

I'm on a databricks runtime 5.4 ML cluster with a scala notebook. I'm essentially trying to reproduce one of the [mleap-xgboost-spark unit tests](https://github.com/combust/mleap/blob/master/mleap-xgboost-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/mleap/XGBoostRegressionModelParitySpec.scala) but it fails on `loadMleapBundle`.

I've ensured that the cluster has mleap-runtime and mleap-xgboost-runtime jars attached. The remaining dependencies should already be part of the databricks runtime.

Are there any scala examples showing how to use the mleap-xgboost integration from within a databricks notebook?

Thanks!

### PR DESCRIPTION
**Pull request by mengxr**
* Update xgboost4j to 0.90, which is compatible with Spark 2.4.
* Include mleap-xgboost-runtime in mleap-databricks-runtime assembly to fix #562 .

Closes #579

### CODE REVIEW

File: project/MleapProject.scala [hunk 1]
```diff
@@ -16,7 +16,6 @@ object MleapProject {
     sparkTestkit,
     spark,
     sparkExtension,
-    tensorflow,
```
Comment (mengxr) [inline project/MleapProject.scala]: already included

**Comment by mengxr** [comment]
cc: @ancasarb @WeichenXu123

**Comment by mengxr** [comment]
@ancasarb Do you mind making 0.15 release after this PR? The xgboost4j upgrade is necessary to work with Spark 2.4.

**Comment by WeichenXu123** [comment]
LGTM!

**Comment by mengxr** [comment]
Ping @ancasarb :)

File: mleap-xgboost-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/mleap/XGBoostRegressionModelParitySpec.scala [hunk 2]
```diff
@@ -36,7 +36,7 @@ class XGBoostRegressionModelParitySpec extends FunSpec
   private val xgboostParams: Map[String, Any] = Map(
     "eta" -> 0.3,
     "max_depth" -> 2,
-    "objective" -> "reg:linear",
+    "objective" -> "reg:squarederror",
```
Comment (ancasarb) [inline mleap-xgboost-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/mleap/XGBoostRegressionModelParitySpec.scala]: just for my understanding, is this change required?

Comment (mengxr) [reply mleap-xgboost-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/mleap/XGBoostRegressionModelParitySpec.scala]: yes, otherwise this test will fail.
